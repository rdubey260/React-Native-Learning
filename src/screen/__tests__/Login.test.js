import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import LoginScreen from '../Login';

const navigate = jest.fn();

const component = (
  <Provider store={store}>
    <LoginScreen navigation={{ navigate }} />
  </Provider>
);

test('render the UI', () => {
  const login = render(component);
  const tree = login.toJSON();
  expect(tree).toMatchSnapshot();
});

test('email input', () => {
  const { getByTestId } = render(component);
  const inputEmail = getByTestId('input-email');
  fireEvent.changeText(inputEmail, 'abc');
  expect(inputEmail.props.value).toBe('abc');
});

test('password input', () => {
  const { getByTestId } = render(component);
  const inputEmail = getByTestId('input-password');
  fireEvent.changeText(inputEmail, 'abc');
  expect(inputEmail.props.value).toBe('abc');
});

test('Login button click', () => {
  const { getByTestId } = render(component);
  const button = getByTestId('login-button');
  fireEvent(button, 'press');
});
