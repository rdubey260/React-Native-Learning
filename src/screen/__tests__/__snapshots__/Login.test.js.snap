// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the UI 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "marginBottom": 20,
      }
    }
  >
    Login user
  </Text>
  <View
    style={
      {
        "flexDirection": "column",
        "marginTop": 10,
      }
    }
  >
    <Text>
      User Name
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        {
          "borderColor": "blue",
          "borderWidth": 2,
          "width": 350,
        }
      }
      testID="input-email"
      value=""
    />
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "marginTop": 10,
      }
    }
  >
    <Text>
      Password
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        {
          "borderColor": "blue",
          "borderWidth": 2,
          "width": 350,
        }
      }
      testID="input-password"
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "green",
        "height": 30,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": 120,
      }
    }
    testID="login-button"
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
